source/extensions/filters/udp/udp_proxy/udp_proxy_filter.h

```diff
@@ -580,7 +580,7 @@ class UdpProxyFilter : public Network::UdpListenerReadFilter,
     const Event::TimerPtr idle_timer_;
 
     UdpProxySessionStats session_stats_{};
-    StreamInfo::StreamInfoImpl udp_session_info_;
+    std::shared_ptr<StreamInfo::StreamInfoImpl> udp_session_info_;
```

Comment:
Why you need to change the stream info as shared_ptr?

Reply:
Otherwise, I can't use the stream info here: https://github.com/envoyproxy/envoy/pull/30736/files#:~:text=access_log%2D%3Elog(nullptr%2C%20nullptr%2C%20nullptr%2C%20*stream_info%2C%20AccessLog%3A%3AAccessLogType%3A%3ANotSet)%3B
Because it will be destructed after active session destruction.

Reply:
is it possible to log before `sessions_.erase(session);`?

Reply:
No, as you can see in the description of the PR: "In this change, moved access log to be after the call of ActiveSession destructor to allow destructors of session filters to run before writing session access log."

Reply:
oh, thanks, I found I thought the whole thing in opposite way. I see now.

May I ask what is the usecase of setting the filter stat in the destructor of session filters?

Also it will be great to have test to cover this case.

Reply:
Added a test.
One use case can be parsing received response trailers from upstream and setting the result in filter state.